src/devices/video/k051316.cpp

```diff
@@ -228,7 +235,17 @@ u8 k051316_device::rom_r(offs_t offset)
 
 void k051316_device::ctrl_w(offs_t offset, u8 data)
 {
+	if (offset == 0x0e)
+	{
+		m_readout_enabled = ~BIT(data, 0);
+		m_flipx_enabled = BIT(data, 1);
+		m_flipy_enabled = BIT(data, 2);
+		if ((data & 0x06) != (m_ctrlram[0x0e] & 0x06))
+			m_tmap->mark_all_dirty();
+	}
```

Comment:
The trouble with adding these extra members is you now have duplicated state.  That’s a path to bugs.

Also, this doesn’t actually work.  You need `BIT(~data, 0)` or `!BIT(data, 0)` – `~BIT(data, 0)` will always be `true` when converted to `bool`.